### PR TITLE
SNOW-75649 Add a flag to control parameter check.

### PR DESCRIPTION
**Pull request by ChTimTsubasa**
### Description
The default behavior is to enable validation;

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: driver_test.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ func setup() (string, error) {
 }
 
 // teardown drops the test schema
-func teardown(s string) error {
```
Comment (ChTimTsubasa) [inline driver_test.go]: IDE's static check complains the s is not used.

File: driver_test.go [hunk 2]
```diff
@@ -1841,12 +1839,12 @@ func TestPingInvalidHost(t *testing.T) {
 		LoginTimeout: 10 * time.Second,
 	}
 
-	url, err := DSN(&config)
```
Comment (ChTimTsubasa) [inline driver_test.go]: IDE's static check complains this variable conflict with the package imported

File: dsn.go [hunk 3]
```diff
@@ -349,16 +369,22 @@ func fillMissingConfigParameters(cfg *Config) error {
 	if strings.Trim(cfg.Application, " ") == "" {
 		cfg.Application = clientType
 	}
+
+	if cfg.OCSPFailOpen == ocspFailOpenNotSet {
+		cfg.OCSPFailOpen = OCSPFailOpenTrue
+	}
```
Comment (smtakeda) [inline dsn.go]: oh, was this missing?

File: dsn.go [hunk 4]
```diff
@@ -349,16 +369,22 @@ func fillMissingConfigParameters(cfg *Config) error {
...
 	if strings.HasSuffix(cfg.Host, defaultDomain) && len(cfg.Host) == len(defaultDomain) {
 		return &SnowflakeError{
 			Number:      ErrCodeFailedToParseHost,
 			Message:     errMsgFailedToParseHost,
 			MessageArgs: []interface{}{cfg.Host},
 		}
 	}
-	if cfg.OCSPFailOpen == ocspFailOpenNotSet {
-		cfg.OCSPFailOpen = OCSPFailOpenTrue
-	}
```
Comment (smtakeda) [inline dsn.go]: ah, ok, you just moved this to above.

**Comment by smtakeda** [comment]
created `prep-1.3.0`

**Comment by smtakeda** [comment]
LGTM. create PR to `prep-1.3.0`

**Pull request merged**